### ISSUE TITLE
[Feature] Share with any external email (without federation)

### ISSUE THREAD

**Issue by freechelmi** [owncloud/core#20074]
Sorry , i'm sure someone logged that already but could not find it :  
### Steps to reproduce
1. Click share on a file
2. enter any email user@domaine.com
3. an error is returned that the federation was not found for domain.com
### Expected behaviour

You should be able to quickly share a file with any email address, a public link would be sent to this email address. 
### Actual behaviour

an error is returned because the domain is not federated.
### Server configuration

**Operating system**: 
any

**Web server:**
 any 

**Database:** 
any
**PHP version:**
any
**ownCloud version:** (see ownCloud admin page)
8.2.0
**Updated from an older ownCloud or fresh install:**
8.1
**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder
```

Enabled:
- activity: 2.1.3
- bookmarks: 0.4
- calendar: 0.8.0
- contacts: 0.5.0.0
- direct_menu: 0.3
- documents: 0.10.2
- external: 1.2
- files: 1.2.0
- files_locking: 
- files_mv: 0.8.1a
- files_pdfviewer: 0.7
- files_sharing: 0.7.0
- files_texteditor: 2.0
- files_trashbin: 0.7.0
- files_versions: 1.1.0
- files_videoviewer: 0.1.3
- firstrunwizard: 1.1
- gallery: 14.2.0
- html5_videoplayer: 1.0
- mail: 0.2.1
- music: 0.3.7
- news: 6.0.6
- notifications: 0.1.0
- ocdownloader: 1.5
- provisioning_api: 0.3.0
- storagecharts2: 2.6.1
- tasks: 0.8
- templateeditor: 0.1
- updater: 0.6
- user_ldap: 0.7.0
  Disabled:
- audios
- chat
- delete_confirmation
- encryption
- files_external
- user_external

**The content of config/config.php:**

<?php
$CONFIG = array (

  'trusted_domains' => 
  array (
    0 => 'drive.ekimia.fr',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'https://drive.ekimia.fr:8091',
  'dbtype' => 'mysql',
  'version' => '8.2.0.12',
  'dbname' => 'ocdb',
  'dbhost' => 'localhost',

  'logtimezone' => 'UTC',
  'installed' => true,
  'mail_from_address' => 'drive',
  'mail_smtpmode' => 'php',
  'mail_domain' => 'ekimia.fr',
  'loglevel' => 1,
  'appstore.experimental.enabled' => true,
  'ldapIgnoreNamingRules' => false,
  'maintenance' => false,
  'theme' => '',
  'trashbin_retention_obligation' => 'auto',
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
);

**Comment by nickvergessen** [comment]
Well if you click on the "share with link" option, you then can share it via email. I see the intention behind your idea, the problem is that emails and federation IDs can be the same value. How should the server blindly guess what you ment?

cc @jancborchardt for "design is how it works"-stuff

**Comment by freechelmi** [comment]
I would say : 

- Check if domain.com is a federated domaine then use 'federation' way of sharing.

- Then just treat is as a normal email, and share as public by sending the link by email, warn the user that anyone with the link might be able to read the file.

- Also the autocomplete should look into the user OC contacts.


Any already opened feature close to this ?

Thanks

**Comment by nickvergessen** [comment]
> Also the autocomplete should look into the user OC contacts.

That is already done as of 8.2 or 8.1, can't remember atm.

**Comment by schiesbn** [comment]
> That is already done as of 8.2 or 8.1, can't remember atm.

But we only look for federated cloud ids, not for normal emails. 
The email field for public link shares should search the address book for email addresses, iirc.

**Comment by jancborchardt** [comment]
This is already accounted for in Sharing 2.0 proposal #19331

> **Multiple link shares**
> Currently only 1 link share is allowed per file. Having multiple makes sense since I might want to share a file with multiple people but not for the same period of time. This could be combined with easily sharing via email so the individual link is automatically sent to the respective person. Individual links can then also be revoked easily based on email/name.

So yes, it should be combined. All the magic like checking for federated cloud ID etc, contact autocompletion should of course be done in the background. I know it’s not easy, but sharing is one of our big selling points so we should do it right.

**Comment by freechelmi** [comment]
I'm quite new in owncloud. 

Do you think we could implement this in a third party app ?

**Comment by schiesbn** [comment]
@freechelmi for ownCloud 9.0 we are working on a new sharing which should more modular. If you are interested you can read about our plans here https://github.com/owncloud/core/issues/19331

Once this is done we could think about implementing this as a additional "share provider" which could be developed as a separate app. Still a problem would be to decide if a address foo@bar is a email address or a cloud id. If we auto-complete the address/id from the address book we could indicate it, but what happens if it is a complete new app/id. Maybe then the share manager would know what to prioritize? cc @rullzer 

Nevertheless, if you are interested please follow the discussion of sharing 2.0 and even contribute if you like.

**Comment by butonic** [comment]
@freechelmi you can have a look at https://github.com/butonic/guests ... feedback is welcome.

**Comment by freechelmi** [comment]
Hi Just upgrade to 9.0.1 , Sharing now propose all the emails from the adress book, but no emails are sent when chossing an email not linked to an owncloud account.

Is this the default wanted behaviour ?

**Comment by jancborchardt** [comment]
cc @rullzer regarding the multiple link share / share to email feature.

**Comment by PVince81** [comment]
guests app: https://marketplace.owncloud.com/apps/guests

**Comment by PVince81** [comment]
please use the guests app
